python/cudf/cudf/tests/test_list.py

```diff
@@ -292,10 +292,12 @@ def test_get_nested_lists():
     assert_eq(expect, got)
 
 
-def test_get_nulls():
-    with pytest.raises(IndexError, match="list index out of range"):
-        sr = cudf.Series([[], [], []])
-        sr.list.get(100)
+def test_get_default():
+    sr = cudf.Series([[1, 2], [3, 4, 5], [6, 7, 8, 9]])
+
+    assert_eq(cudf.Series([cudf.NA, 5, 8]), sr.list.get(2))
+    assert_eq(cudf.Series([cudf.NA, 5, 8]), sr.list.get(2, default=cudf.NA))
+    assert_eq(cudf.Series([0, 5, 8]), sr.list.get(2, default=0))
```

Comment:
1. Is there implicit casting if the default is not the same type as the column?
2. How are negative indices handled? I don't see any existing tests for negative indices above.
3. Can we test strings with a default value as well, for better type coverage?

Below is a suggestion with my expectations for those behaviors (though I'm not sure about my expectations for casting - maybe a dtype check and raising an error is appropriate?).
```suggestion
    assert_eq(cudf.Series([0, 5, 8]), sr.list.get(2, default=0))
    assert_eq(cudf.Series([0.5, 5., 8.]), sr.list.get(2, default=0.5))
    assert_eq(cudf.Series([2, 5, 9]), sr.list.get(-1))
    
    string_sr = cudf.Series([["apple", "banana"], ["carrot", "daffodil", "elephant"]])
    assert_eq(cudf.Series(["default", "elephant"]), string_sr.list.get(1, default="default"))
```

Reply:
1. The default value just goes through the `fillna()` machinery, which does not do implicit casting. I think we should keep that behavior? The resulting `TypeError` seems good enough.
2. Negative indices are handled at the libcudf level by the "indexalator", AFAIR - again, we should add a test.
3. Yes!

I  added your suggested tests. I think for the last one you mean to do `.get(2, default="default")`